book/src/tutorial/qt-gui/qquickpainteditem.md

```diff
@@ -0,0 +1,211 @@
+You can read up on CXX in the [CXX book][CXX].
+
+To create our `QQuickPaintedItem` subclass, we first need to include the appropriate C++ headers,
+then declare the types we want to use from Rust.
+For our first example, we'll only set the `fillColor` and not actually paint anything using the `QPainter`.
+
+✅ Create a new `#[cxx_qt::bridge]` module
+
+It should include the following declarations/includes:
+* ✅ An import of `QColor` from cxx-qt-lib [(see the CXX-Qt documentation)](https://docs.rs/cxx-qt-lib/latest/cxx_qt_lib/struct.QColor.html)
+* ✅ An include to the `QQuickPaintedItem` header
+* ✅ An include for the `QPainter` header, as well as a declaration of the `QPainter` type.
+
+```rust
+#[cxx_qt::bridge(cxx_file_stem = "image_painter")]
+pub mod qobject {
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/qcolor.h");
+        type QColor = cxx_qt_lib::QColor;
+
+        include!(<QtQuick/QQuickPaintedItem>);
+    }
+
+    unsafe extern "C++" {
+        include!(<QtGui/QPainter>);
+        type QPainter;
+    }
+}
+```
+> Take a look at the [CXX documentation][CXX] and the [CXX-Qt-lib](https://docs.rs/cxx-qt-lib/latest/cxx_qt_lib/index.html) documentation.
+>
+> ❓ Is `QColor` an opaque or a trivial type? What does this mean when using QColor in Rust?\
+> ❓ What's the difference between declaring a `type QPainter` vs. `type QPainter = ...;`?\
+> ❓ What's the difference between `unsafe extern "C++"` and `extern "C++"`?
+
+Now that we have our types ready to go, we can declare our new `QQuickPaintedItem` subclass.
+
+With CXX-Qt this means following these steps:
+* ✅ Create a new QObject subclass using `type X = super::Y;` with the `#[qobject]` attribute
+    * This tells CXX-Qt to create a new QObject subclass of the name `X`.
+        This class will contain an instance of the `super::Y` Rust struct which contains the actual data (nothing for now).
+* ✅ Tell CXX-Qt to use `QQuickPaintedItem` as our base class using `#[base="..."]`
+* ✅ Declare the necessary functions
+    * ✅ Tell Rust that `setFillColor` is inherited from `QQuickPaintedItem`
+    * ✅ Declare that `paint` should override the base class implementation and is implemented in Rust.
+* ✅ Tell CXX-Qt that we want a default-constructor for ImagePainter.
+    * We'll implement this constructor later in Rust.
+```rust
+#[cxx_qt::bridge(cxx_file_stem = "image_painter")]
+pub mod qobject {
+    // ...
+
+    unsafe extern "RustQt" {
+        #[qml_element]
+        #[base = "QQuickPaintedItem"]
+        #[qobject]
+        type ImagePainter = super::ImagePainterRust;
+
+        #[inherit]
+        #[rust_name = "set_fill_color"]
+        fn setFillColor(self: Pin<&mut ImagePainter>, color: &QColor);
+
+        #[cxx_override]
+        unsafe fn paint(self: Pin<&mut ImagePainter>, painter: *mut QPainter);
+    }
+
+    impl cxx_qt::Constructor<()> for ImagePainter {}
+}
+```
+> Some food for thought:
+>
+> ❓ What does the `#[qml_element]` attribute do?\
```

Comment:
do we need the `\` again here ? maybe it's doing something in markdown ?

Reply:
Yes, it changes the formatting.

Usually two consecutive lines in the mdbook flavor of Markdown will be merged into a paragraph:

i.e.:
```markdown
hello
world
```
renders as:

hello world

whilst:
```markdown
hello\
world
```

renders as:

hello\
world

I think Github doesn't do this though :thinking: